src/main/java/dk/bec/bookanything/service/FacilityService.java

```diff
@@ -0,0 +1,73 @@
+package dk.bec.bookanything.service;
+
+import dk.bec.bookanything.dto.DayOpenReadDto;
+import dk.bec.bookanything.dto.FeatureReadDto;
+import dk.bec.bookanything.mapper.DayOpenMapper;
+import dk.bec.bookanything.mapper.FeatureMapper;
+import dk.bec.bookanything.model.AddressEntity;
+import dk.bec.bookanything.model.FacilityEntity;
+import dk.bec.bookanything.model.FacilityTypeEntity;
+import dk.bec.bookanything.repository.FacilityRepository;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class FacilityService {
+
+    private final FacilityRepository facilityRepository;
+    private final FeatureMapper featureMapper;
+    private final DayOpenMapper dayOpenMapper;
+
+    public FacilityService(FacilityRepository facilityRepository, FeatureMapper featureMapper, DayOpenMapper dayOpenMapper) {
+        this.facilityRepository = facilityRepository;
+        this.featureMapper = featureMapper;
+        this.dayOpenMapper = dayOpenMapper;
+    }
+
+    public Optional<FacilityEntity> getFacilityById(Long id)
+    {
+      return facilityRepository.findById(id);
+    }
+
+    public Optional<List<FeatureReadDto>> getFeaturesForFacility(Long id)
+    {
+        return getFacilityById(id).map(entity -> entity.getFeatureEntities().stream()
+                .map(featureMapper::mapFeatureEntityToDto
+        ).collect(Collectors.toList()));
+    }
+    public Optional<List<DayOpenReadDto>> getDaysOpenForFacility(Long id)
+    {
+        return getFacilityById(id).map(entity -> entity.getDayOpenList().stream()
+                .map(dayOpenMapper::mapDayOpenEntityToReadDto).collect(Collectors.toList()));
+    }
+
+    @Transactional
+    public Optional<FacilityEntity> createFacility(FacilityEntity facilityEntity) {
+        return Optional.of( facilityRepository.save(facilityEntity));
+    }
+
+    public Optional<FacilityEntity> updateFacility(Long id, FacilityEntity newFacility) {
+        if(getFacilityById(id).isPresent())
```

Comment:
Don't leave if statements without brackets. So if event one line is after "if" statement, lets put that into a bracket. FE:
```
if(getFacilityById(id).isPresent()){
     return Optional.of(facilityRepository.save(newFacility));
}
```